flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/library/HITSAlgorithm.java

```diff
@@ -243,18 +255,22 @@ public void sendMessages(Vertex<K, Tuple2<DoubleValue, DoubleValue>> vertex) {
 				iterationValueSum = Math.sqrt(((DoubleValue) getPreviousIterationAggregate("updatedValueSum")).getValue());
 			}
 			for (Edge<K, EV> edge : getEdges()) {
-				K messageSource = getSuperstepNumber() % 2 == 1 ? edge.getSource() : edge.getTarget();
-				K messageTarget = getSuperstepNumber() % 2 == 1 ? edge.getTarget() : edge.getSource();
-				double messageValue = getSuperstepNumber() % 2 == 1 ? vertex.getValue().f0.getValue() : vertex.getValue().f1.getValue();
-
-				if (!messageTarget.equals(vertex.getId())) {
-					if (getSuperstepNumber() != maxIteration) {
-						sendMessageTo(messageTarget, messageValue / iterationValueSum);
-
-						// in order to make every vertex updated
-						sendMessageTo(messageSource, 0.0);
+				if (getSuperstepNumber() != maxIteration) {
+					if (getSuperstepNumber() % 2 == 1) {
+						if (edge.getValue().equals("Authority")) {
+							sendMessageTo(edge.getTarget(), vertex.getValue().f0.getValue() / iterationValueSum);
+						}
 					} else {
-						sendMessageTo(messageSource, iterationValueSum);
+						if (edge.getValue().equals("Hub")) {
+							sendMessageTo(edge.getTarget(), vertex.getValue().f1.getValue() / iterationValueSum);
+						}
+					}
+					
+					// make all the vertices be updated
+					sendMessageTo(edge.getSource(), 0.0);
```

Comment:
Sorry, this line should be moved. :)
In the previous commit without edge label, i use the same edge for hub and authority updating and this line would keep every vertex updating. But in the newest implementation with edge label, we can drop this because the added edges can do the same work.